CONTRIBUTING.md

```diff
@@ -1,12 +1,134 @@
+    - [Adding new dependency](#adding-new-dependency)
+
+Below you'll find instructions on how to contribute to UMF, either with code changes
+or issues. All contributions are most welcome!
+
+## Opening new issues
+
+Please log bugs or suggestions as [GitHub issues](https://github.com/oneapi-src/unified-memory-framework/issues).
+
+When reporting a bug remember about the details, at least:
+- version of UMF (hash of a commit or a tag),
+- OS and kernel,
+- compiler.
+
+When opening a new issue you can pick a predefined type of issue. Please follow
+the template and fill in all the information. If your query doesn't match any of
+the proposed types, just pick a general issue with no template.
+
+## Submitting Pull Requests
+
+We take outside code contributions to UMF through GitHub pull requests.
+You must first create your own fork of the project and submit your changes to a branch.
+You can then raise a Pull Request targeting `oneapi-src/unified-memory-framework:main`.
+Please try to keep commits neat and in order - please squash your commits to
+include only relevant ones (no "fixes after review" or similar).
+
+### Building and testing
+
+Building commands can be found in the top-level Readme file - section
+["Build"](./README.md#build).
+
+Before committing you should test locally if all tests and checks pass.
+When project is built, enter the build directory and execute:
+
+```bash
+$ ctest --output-on-failure
+```
+
+Any test's failure will produce error log.
+
+To enable additional checks (including `-Werror` / `/WX` compilation flag), switch on CMake flag
+`UMF_DEVELOPER_MODE`. To read more about all available CMake options please see
+["CMake standard options"](./README.md#cmake-standard-options) section in the top-level Readme.
+
+### Code style
+
+We use `clang-format` to verify and apply code style changes. To see all rules we require,
+please take a look at `.clang-format` file in the root directory of this repository.
+
+To enable code style checks and re-formatting, CMake option `UMF_FORMAT_CODE_STYLE` has to
+be switched on. You'll then have two additional CMake targets available.
+
+To verify correct coding style of your changes execute (assuming `build` is your build directory):
+
+```bash
+$ cmake -B build -DUMF_FORMAT_CODE_STYLE=ON
+$ cmake --build build --target clang-format-check
+```
+
+We run this check in our Continuous Integration (CI). So, if any issues were found,
+the Pull Request will be blocked from merging. To apply proper formatting (meaning,
+to fix the issues) execute the second available CMake target - run a command:
+
+```bash
+$ cmake --build build --target clang-format-apply
+
+# Remember to review introduced changes
+```
+
+**NOTE**: We use specific clang-format version - **15.0** is required. It can be installed,
+e.g., with command: `python -m pip install clang-format==15.0.7`.
+
+### When my PR is merged?
+
+To get your Pull Request (PR) merged you have to fulfill several requirements.
```

Comment:
I'm not an expert but I think this could sound better:

To get your Pull Request (PR) to be merged, you must meet several requirements - the basic are:
- The project compiles correctly,
- ....

Reply:
This way is less fun 😄😉 Rewritten.